xlators/cluster/afr/src/afr-open.c

```diff
@@ -311,43 +441,17 @@ afr_fix_open(fd_t *fd, xlator_t *this)
     ret = loc_path(&local->loc, NULL);
     if (ret < 0)
         goto out;
-
     local->fd = fd_ref(fd);
     local->fd_ctx = fd_ctx;
+    local->strict_locks = 0;
 
-    local->call_count = call_count;
-
-    gf_msg_debug(this->name, 0, "need open count: %d", call_count);
-
-    for (i = 0; i < priv->child_count; i++) {
-        if (!need_open[i])
-            continue;
-
-        if (IA_IFDIR == fd->inode->ia_type) {
-            gf_msg_debug(this->name, 0, "opening fd for dir %s on subvolume %s",
-                         local->loc.path, priv->children[i]->name);
-
-            STACK_WIND_COOKIE(frame, afr_openfd_fix_open_cbk, (void *)(long)i,
-                              priv->children[i],
-                              priv->children[i]->fops->opendir, &local->loc,
-                              local->fd, NULL);
-        } else {
-            gf_msg_debug(this->name, 0,
-                         "opening fd for file %s on subvolume %s",
-                         local->loc.path, priv->children[i]->name);
-
-            STACK_WIND_COOKIE(frame, afr_openfd_fix_open_cbk, (void *)(long)i,
-                              priv->children[i], priv->children[i]->fops->open,
-                              &local->loc, fd_ctx->flags & (~O_TRUNC),
-                              local->fd, NULL);
-        }
-
-        if (!--call_count)
-            break;
+    ret = synctask_new(this->ctx->env, afr_do_fix_open, afr_fix_open_cbk, frame,
```

Comment:
This is not a good idea. Each synctask takes 20MB stack. In VM workload kind of scenarios we will get too many fix-open situation increasing the mem-usage which could lead to OOMkill.

Reply:
@pranithk In that case is it okay to block the FOP which calls afr_fix_open() using AFR_ONLIST for the getlk and open? Or should we stack wind the getlk and resume the opening logic in the cbk?

Reply:
Whatever logic we have to add, we shouldn't be blocking.

Reply:
Done.